packages/job-worker/src/playout/lock.ts

```diff
@@ -93,11 +95,15 @@ export async function runWithPlaylistCache<TRes>(
 	try {
 		const res = await fcn(fullCache)
 		logger.silly('runWithPlaylistCache: saveAllToDatabase')
-		await fullCache.saveAllToDatabase()
+		await fullCache.saveAllToDatabase(existingTransaction)
 
 		return res
 	} catch (err) {
-		fullCache.discardChanges()
+		try {
+			fullCache.discardChanges()
+		} catch (e) {
+			// Ignore
```

Comment:
Should this at the very least log some information that this threw an error? It shouldn't throw, generally, right?

Reply:
I hit this because `fcn()` a few lines above threw an error. And then this threw an error which hid the real error from me.
Reading into it, I agree that discardChanges shouldnt be throwing. But also, the discardChanges call here is rather pointless. The discard is reverting its internal state back to before the job was run, and then verifying that it did that correctly. As far as I can tell, this is simply costing us time in doing some clones, just before we discard the cache.
I'll review all the places we call discardChanges, and replace/remove the ones that seem odd like this.